src/main/java/mate/academy/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,227 @@
+package mate.academy.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.UserDao;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lb.Dao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+
+    @Override
+    public User create(User user) {
+        String insertUser = "INSERT INTO users (name, login, password) VALUES (?, ?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertUserStatement = connection
+                        .prepareStatement(insertUser, Statement.RETURN_GENERATED_KEYS)) {
+            insertUserStatement.setString(1, user.getName());
+            insertUserStatement.setString(2, user.getLogin());
+            insertUserStatement.setString(3, user.getPassword());
+            insertUserStatement.executeUpdate();
+            ResultSet resultSet = insertUserStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long userId = resultSet.getLong(1);
+                user.setId(userId);
+                insertUserRoles(user, connection);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creating user " + user
+                    + " is failed", e);
+        }
+        return user;
```

Comment:
does user that you return has ids for his roles, if yes, when did you set them ?